src/pages/workspace/accounting/qbo/export/QuickbooksCompanyCardExpenseAccountSelectCardPage.tsx

```diff
@@ -0,0 +1,107 @@
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as Connections from '@libs/actions/connections';
+import Navigation from '@navigation/Navigation';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import FeatureEnabledAccessOrNotFoundWrapper from '@pages/workspace/FeatureEnabledAccessOrNotFoundWrapper';
+import type {WithPolicyProps} from '@pages/workspace/withPolicy';
+import withPolicyConnections from '@pages/workspace/withPolicyConnections';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+
+type CardListItem = ListItem & {
+    value: ValueOf<typeof CONST.QUICKBOOKS_EXPORT_COMPANY_CARD>;
+};
+type CardsSection = SectionListData<CardListItem, Section<CardListItem>>;
+type Card = {name: string; id: ValueOf<typeof CONST.QUICKBOOKS_EXPORT_COMPANY_CARD>};
+
+function QuickbooksCompanyCardExpenseAccountSelectCardPage({policy}: WithPolicyProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const policyID = policy?.id ?? '';
+    const {exportCompanyCard, syncLocations} = policy?.connections?.quickbooksOnline?.config ?? {};
+    const isLocationEnabled = Boolean(syncLocations && syncLocations !== CONST.INTEGRATION_ENTITY_MAP_TYPES.NONE);
+
+    const defaultCards = useMemo<Card[]>(
+        () => [
+            {
+                name: translate(`workspace.qbo.creditCard`),
+                id: CONST.QUICKBOOKS_EXPORT_COMPANY_CARD.CREDIT_CARD,
+            },
+            {
+                name: translate(`workspace.qbo.debitCard`),
+                id: CONST.QUICKBOOKS_EXPORT_COMPANY_CARD.DEBIT_CARD,
+            },
+            {
+                name: translate(`workspace.qbo.vendorBill`),
+                id: CONST.QUICKBOOKS_EXPORT_COMPANY_CARD.VENDOR_BILL,
+            },
+        ],
+        [translate],
+    );
+    const cards = useMemo<Card[]>(() => (isLocationEnabled ? defaultCards.slice(0, -1) : defaultCards), [isLocationEnabled, defaultCards]);
+
+    const data = useMemo<CardListItem[]>(
+        () =>
+            cards.map((card) => ({
+                value: card.id,
+                text: card.name,
+                keyForList: card.name,
+                isSelected: card.id === exportCompanyCard,
+            })),
+        [cards, exportCompanyCard],
+    );
+
+    const sections = useMemo<CardsSection[]>(() => [{data}], [data]);
+
+    const onSelectRow = useCallback(
+        (row: CardListItem) => {
+            if (row.value !== exportCompanyCard) {
+                Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD, row.value);
+                Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD_ACCOUNT);
```

Comment:
Yeah - but here we need to set one value - and clean up already existed one from other field @s77rt do we have some examples?

Reply:
I saw something like this one:
<img width="733" alt="Screenshot 2024-04-24 at 19 21 16" src="https://github.com/Expensify/App/assets/28352309/f49d0a48-008c-4c24-90de-60e897f058fc">
When Onyx is reseting.. but not sure if that works
Also i can also move this logic to useEffect, based on prev stored card item and new one, but probably it looks like overkill. Maybe it can be done on BE side @hayata-suenaga - if user change company card - BE automatically clean up selected Account/Vendor. Will it be possible?

Reply:
@aldo-expensify, if we're to follow the 1:1:1 pattern, I think we can create a new Web-E command. What do you think?

Reply:
So when you call `CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD`, we have to always clear `CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD_ACCOUNT`?

Reply:
Yes, we have to follow 1:1:1, and I don't think we should go around with a `useEffect`. I see two options:

Option 1: We keep using `UpdatePolicyConnectionConfig` and we change the backend to clear `CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD_ACCOUNT` if `CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD` is being set. That would mean that we also have to add this exception in `Connections.updatePolicyConnectionConfig` to create the optimistic data correctly

Option 2: Add a new command

Reply:
@aldo-expensify agree with Option 2 - just having one additional command probably will be better solution

Reply:
internal follow up issue -> https://github.com/Expensify/Expensify/issues/390731